### DISCUSSION TITLE
Resource Sub-Folders & Organization

### DISCUSSION THREAD

**Discussion by lyimu** [AscensionGameDev/Intersect-Engine#1482] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm reaching the point where my mapping speeds and developing speeds are throttled by a lack of organization in my project.
I try to keep everything as organized as possible with tilesets and other assets starting with "indoors_**tilesettype**_etc,etc"but there is only so much I can do.

**Describe the solution you'd like**
Allow the Intersect Engine to recognize sub-folders inside of the resource folders and categorize them when selecting tiles, sprites, items, paperdolls etc.

**Image Context**
Categorizing your tilesets into sub-folders would look something like this in the tilesets folder.
![Screenshot_2](https://user-images.githubusercontent.com/25600834/97892300-b0d80380-1ce4-11eb-92bd-94088a34d530.png)

You would then be able to select the category and then select the tileset inside that category like this.
![image](https://user-images.githubusercontent.com/25600834/97892648-23e17a00-1ce5-11eb-93be-6e44169b5d2a.png)
 (_just imagine the category box is at the top though_)

**Alternative Workarounds**
Say the idea above doesn't work for some people, perhaps instead of creating sub-folders yourself in your resource folders just naming your tilesets in a certain way would do it for you.

Like this:
![image](https://user-images.githubusercontent.com/25600834/97893359-07920d00-1ce6-11eb-84de-63a4bb994aff.png)
(_Of course make this a optional feature via config as it could conflict with existing projects_)

**Comment by WeylonSantana** [comment]
I'm all for it! if there was a way to do it dynamically. But I would like to add that this should be for b8 only
